src/System.Text.Utf8String/System/Text/Utf8String.CodePointEnumerable.cs

```diff
@@ -4,18 +4,22 @@
 
 namespace System.Text.Utf8
 {
-    partial struct Utf8String
+    partial class Utf8String
     {
         public ref struct CodePointEnumerable 
         {
-            private ReadOnlySpan<byte> _buffer;
+            private byte[] _buffer;
 
             public CodePointEnumerable(byte[] bytes, int index, int length)
             {
-                _buffer = new ReadOnlySpan<byte>(bytes, index, length);
+                _buffer = new byte[length];
```

Comment:
Should this be `new byte[index - length]` and then we copy only the subset of the bytes from the input array?